mdbook-tera-backend/src/tera_renderer/renderer.rs

```diff
@@ -0,0 +1,140 @@
+    /// Create a new `Renderer` from the `RenderContext` and `Tera` template.
+    pub(crate) fn new(ctx: RenderContext, tera_template: Tera) -> Result<Renderer> {
+        let mut renderer = Renderer {
+            ctx: Arc::new(ctx),
+            tera_template,
+        };
+        renderer
+            .tera_template
+            .register_function("get_context", renderer.create_get_context_function());
+        Ok(renderer)
+    }
+
+    /// Render the book. This goes through the output of the HTML renderer
+    /// by considering all the output HTML files as input to the Tera template.
+    /// It overwrites the preexisting files with their Tera-rendered version.
+    pub(crate) fn render_book(&mut self) -> Result<()> {
+        let dest_dir = self
+            .ctx
+            .destination
+            .parent()
+            .unwrap()
+            .join("html")
+            .to_owned();
+        if !dest_dir.is_dir() {
+            return Err(anyhow!(
+                "{dest_dir:?} is not a directory. Please make sure the HTML renderer is enabled."
+            ));
+        }
+        self.render_book_directory(&dest_dir)
+    }
+
+    /// Create the `get_context` function for the `Tera` template, a helper that allows retrieving values from `ctx`.
+    fn create_get_context_function(&self) -> impl tera::Function {
+        let ctx_rc = Arc::clone(&self.ctx);
+        move |args: &HashMap<String, serde_json::value::Value>| -> tera::Result<tera::Value> {
+            let key = args
+                .get("key")
+                .ok_or_else(|| tera::Error::from(format!("No key argument provided")))?
+                .as_str()
+                .ok_or_else(|| {
+                    tera::Error::from(format!("Key has invalid type, expected string"))
+                })?;
+            let value = ctx_rc
+                .config
+                .get(key)
+                .ok_or_else(|| tera::Error::from(format!("Could not find key {key} in config")))?;
+            let value = to_value(value)?;
+            Ok(value)
+        }
+    }
+
+    /// Render the book directory located at `path` recursively.
+    fn render_book_directory(&mut self, path: &Path) -> Result<()> {
+        for entry in path.read_dir()? {
+            let entry = entry?;
+            let path = entry.path();
+            if path.is_dir() {
+                self.render_book_directory(&path)?;
+            } else {
+                self.process_file(&path)?;
+            }
+        }
+        Ok(())
+    }
+
+    /// Reads the file at `path` and renders it.
+    fn process_file(&mut self, path: &Path) -> Result<()> {
+        if path.extension().unwrap_or_default() != "html" {
+            return Ok(());
+        }
+        let file_content = std::fs::read_to_string(path)?;
+        let output = self.render_file_content(&file_content, path)?;
+        let mut output_file = fs::File::create(path)?;
+        output_file.write_all(output.as_bytes())?;
+        Ok(())
```

Comment:
Please try with [std](https://doc.rust-lang.org/stable/std/index.html)::[fs](https://doc.rust-lang.org/stable/std/fs/index.html)::[write](https://doc.rust-lang.org/stable/std/fs/fn.write.html#):

```suggestion
        fs::write(path, output)
```